[editorial] Setup Prettier and run it on some files

chalin opened the pull request:
- A first step towards #189.
- We can then progressively add files, and/or tune Prettier options.

/cc @jsuereth @joaopgrassi

joaopgrassi commented:
Can we run it on one of the other files (maybe one of the largest ones?) to see what kind of changes would happen before we decide to start using it? I'm generally in favor for adding such things to take away debates about style :)

chalin commented:
Sure @joaopgrassi. I've added all `/docs/cloud*` files to the list of processed files. Does that help?

Note the required use of `<!-- prettier-ignore-start/end -->` before autogenerated semconv tables.

chalin commented on package.json:
```diff
@@ -2,6 +2,10 @@
+    "prettier": "^3.0.0"
+  },
+  "prettier": {
+    "proseWrap": "preserve"
```
Note that eventually we can to change this option to `always`. That makes for a better experience viewing PR diffs in GitHub. For now, to minimize changes, we should keep `preserve`.

chalin commented:
FYI, if you select the "rich diff" view, you'll see that there are no changes for most files, e.g.:

> <img width="688" alt="image" src="https://github.com/open-telemetry/semantic-conventions/assets/4140793/baa6f614-9b04-4c27-b613-58f3526cff55">

chalin commented on docs/cloudevents/cloudevents-spans.md:
```diff
@@ -39,7 +40,7 @@ document.
 For such a scenario, the traditional parent-child trace model works well.
 However, CloudEvents are also used in distributed systems where an event
-can go through many [hops](https://en.wikipedia.org/wiki/Hop_(networking))
+can go through many [hops](<https://en.wikipedia.org/wiki/Hop_(networking)>)
```
No, only in cases where a URL contains characters that _might_confuse markdown processors: in this case, the presence of parentheses is that triggers this adjustment.

joaopgrassi commented:
Do we need to maybe add instructions on how to "invoke" prettier to apply changes/fixes locally? Thinking on how people should address their changes after this is merged. Or do you plan to do this as a follow up? Also - do we want to add checks as a GitHub action?

chalin commented:
> Do we need to maybe add instructions on how to "invoke" prettier to apply changes/fixes locally?

Done. I've added scripts and updated the CONTRIBUTING instructions.

(I haven't yet added a CI check.)

jsuereth commented on CONTRIBUTING.md:
````diff
@@ -61,7 +62,7 @@ Semantic conventions have some autogenerated components and additionally can do
 
 ```bash
-make fix
+npm run fix
````
Can we keep `make` as the entry point, and just have it call `npm` ?

chalin replied:
Certainly. Done, PTAL.
